apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -104,35 +109,37 @@ export default class GenerateAction extends BaseRushAction {
       fsx.unlinkSync(tempShrinkwrapFilename);
     }
 
-    if (isLazy) {
-      console.log(colors.green(
-        `${os.EOL}Rush is running in "--lazy" mode. ` +
-        `You will need to run a normal "rush generate" before committing.`));
+    if (isLazy || this.rushConfiguration.packageManager === 'pnpm') {
+      if (this.rushConfiguration.packageManager === 'npm') {
+        console.log(colors.green(
+          `${os.EOL}Rush is running in "--lazy" mode. ` +
+          `You will need to run a normal "rush generate" before committing.`));
+      }
 
       // Do an incremental install
```

Comment:
> // Do an incremental install [](start = 5, length = 29)

What if the PNPM install is corrupted?  We probably need a "--clean" option to force a non-incremental install, right?

Reply:
PNPM shouldn't be corrupted, remember that incremental installs check the install flag to make sure we don't need to clean out the cache and node_modules. That being said, we probably should have a way to force that behavior.

---
In reply to: [153377641](https://github.com/Microsoft/web-build-tools/pull/449#discussion_r153377641) [](ancestors = 153377641)